Dockerfile

```
@@ -78,11 +78,14 @@ RUN mkdir -p /etc/sysconfig/slurm \
         /var/lib/slurmd/qos_usage \
         /var/lib/slurmd/fed_mgr_state 
 
-ARG PUBLIC_KEY
+RUN ssh-keygen -q -t rsa -N '' -f /container_root_pubkey
 
-# setup login to ssh 
+# setup login to ssh
 RUN mkdir -p /root/.ssh/ \
-    && echo "$PUBLIC_KEY" >> /root/.ssh/authorized_keys \
+    && chmod 700 /container_root_pubkey \
+    && echo $(cat /container_root_pubkey.pub) >> /root/.ssh/authorized_keys \
+    && chmod 700 /root/.ssh/ \
+    && chmod 700 /root/.ssh/authorized_keys \
```

Make `chmod` recursive, so you can remove one line.

```suggestion
    && chmod -R 700 /root/.ssh/ \
```